### PR TITLE
Add a latch lever and contact sensor

### PR DESCRIPTION
**Pull request by ibrahiminfinite**
This PR adds a rod with contact sensor as the mock latching mechanism of the robot
Contact data is published on `/latch_lever/contact`
Cart name date is published on `latch_lever/cart_name`

### CODE REVIEW

File: include/nav2_gazebo/cart_name_extractor.hpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#include <rclcpp/rclcpp.hpp>
+#include <ros_gz_interfaces/msg/contacts.hpp>
+
+namespace nav2_gazebo
+{
+  class CartContactDetector : public rclcpp::Node
+  {
+    public:
+
+        CartContactDetector();
```
Comment (AndyZe) [inline include/nav2_gazebo/cart_name_extractor.hpp]: Indentation should only be 2 spaces

File: src/cart_name_extractor.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  }
+
+  void CartContactDetector::contactCallback(const ros_gz_interfaces::msg::Contacts::SharedPtr msg)
+  {
+    auto contact = msg.contacts[0]; // ros_gz_interfaces/msgs/Contacts
+    auto other_entity = contact.collision2; // ros_gz_interfaces/msgs/Entity
+    cart_name_ = other_entity.name; // string
+    
+    publishCartName();
```
Comment (AndyZe) [inline src/cart_name_extractor.cpp]: How about getting rid of `cart_name_`, you could pass it as an arg to `publishCartName()` like this (I think...)

```suggestion
    cart_name_ = other_entity.name; // string
    
    publishCartName(other_entity.name);
```

File: include/nav2_gazebo/cart_name_extractor.hpp (hunk 1 above)
Comment (ibrahiminfinite) [reply include/nav2_gazebo/cart_name_extractor.hpp]: Fixed

File: src/cart_name_extractor.cpp (hunk 2 above)
Comment (ibrahiminfinite) [reply src/cart_name_extractor.cpp]: Eliminated the unnecessary storing of cart name

**Pull request merged**